src/v/kafka/server/handlers/describe_groups.cc

```diff
@@ -17,51 +17,57 @@
 #include "model/namespace.h"
 #include "resource_mgmt/io_priority.h"
 
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/when_all.hh>
+
 namespace kafka {
 
 void describe_groups_response::encode(
   const request_context& ctx, response& resp) {
     data.encode(resp.writer(), ctx.header().version);
 }
 
-struct describe_groups_ctx {
-    request_context rctx;
-    describe_groups_request request;
-    describe_groups_response response;
-    ss::smp_service_group ssg;
-
-    describe_groups_ctx(
-      request_context&& rctx,
-      describe_groups_request&& request,
-      ss::smp_service_group ssg)
-      : rctx(std::move(rctx))
-      , request(std::move(request))
-      , ssg(ssg) {}
-};
-
 template<>
 ss::future<response_ptr> describe_groups_handler::handle(
   request_context ctx, ss::smp_service_group ssg) {
     describe_groups_request request;
     request.decode(ctx.reader(), ctx.header().version);
     klog.trace("Handling request {}", request);
 
-    return ss::do_with(
-      describe_groups_ctx(std::move(ctx), std::move(request), ssg),
-      [](describe_groups_ctx& octx) {
-          return ss::parallel_for_each(
-                   octx.request.data.groups.begin(),
-                   octx.request.data.groups.end(),
-                   [&octx](kafka::group_id id) {
-                       return octx.rctx.groups()
-                         .describe_group(std::move(id))
-                         .then([&octx](described_group g) {
-                             octx.response.data.groups.push_back(std::move(g));
-                         });
-                   })
-            .then(
-              [&octx] { return octx.rctx.respond(std::move(octx.response)); });
+    auto unauthorized_it = std::partition(
+      request.data.groups.begin(),
+      request.data.groups.end(),
+      [&ctx](const group_id& id) {
+          return ctx.authorized(security::acl_operation::describe, id);
       });
+
+    std::vector<group_id> unauthorized(
```

Comment:
i guess this does `end()-begin()` and reserve or should we use the std::move( begin(), end(), std:::insert( new_collection))`

it probably does reserve. though this isn't in the hotpath

Reply:
this was a `nit`

Reply:
there are a lot of constructors but it looks like this one does reserve

![image](https://user-images.githubusercontent.com/242417/113964132-baa7ea80-97df-11eb-816c-77d73aff4eea.png)

Reply:
smart! makes sense.